java/src/main/java/com/google/appengine/tools/pipeline/PipelineServiceFactory.java

```diff
@@ -14,18 +14,65 @@
 
 package com.google.appengine.tools.pipeline;
 
+import com.google.appengine.api.utils.SystemProperty;
 import com.google.appengine.tools.pipeline.impl.PipelineServiceImpl;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.datastore.Datastore;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Base64;
 
 /**
  * A factory for obtaining instances of {@link PipelineService}
- * 
+ *
  * @author rudominer@google.com (Mitch Rudominer)
  */
 public final class PipelineServiceFactory {
   private PipelineServiceFactory() {
   }
 
+  /**
+   * @return PipelineService using application defaults for GCP infra needed underneath
+   */
+  @Deprecated //coupled to AppEngine
   public static PipelineService newPipelineService() {
-    return new PipelineServiceImpl();
+    return newPipelineService(SystemProperty.applicationId.get());
+  }
+
+  /**
+   * @param projectId GCP project under which pipelines will execute
+   * @return PipelineService that will execute pipelines in specific project, auth'd by application default credentials
+   */
+  public static PipelineService newPipelineService(String projectId) {
+    try {
+      GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+      return new PipelineServiceImpl(projectId, credentials);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to get default credentials", e);
+    }
+  }
+
+  public static PipelineService newPipelineService(String projectId, Datastore datastore) {
+    return new PipelineServiceImpl(projectId, datastore);
+  }
+
+  /**
+   *
+   * @param projectId GCP project under which pipelines will execute
+   * @param base64EncodedServiceAccountKey service account key used to authenticate for access to that GCP project
+   *                                       (need not be from same project, but must have datastore/task queue perms)
+   * @return PipelineService that will execute pipelines in specific project, auth'd by serviceAccountKey
+   */
+  public static PipelineService newPipelineService(String projectId, String base64EncodedServiceAccountKey) {
```

Comment:
this, or of the defaults, is what we would use for production; I originally though this would work for testing, before seeing that there's no credential used when speaking with emulator.